spec/models/lti_deployment_spec.rb

```diff
@@ -143,4 +143,167 @@
+                                                                   ] }]
+                                                   }.to_json)
+      stub_request(:post, "#{lti_line_item.lti_line_item_id}/scores").with(headers:
+                                                                            { Authorization: 'Bearer access_token' },
+                                                                           body: hash_including({
+                                                                             scoreGiven: /\d+\.?\d+/,
+                                                                             scoreMaximum: assessment.max_mark.to_s,
+                                                                             activityProgress: 'Completed',
+                                                                             gradingProgress: 'FullyGraded'
+                                                                           })).to_return(status: :success)
+    end
+    it 'does send grades' do
+      expect(lti_deployment.create_grades(assessment)).not_to be_empty
+    end
+  end
+  describe '#get_assignment_grades' do
+    let(:course) { create :course }
+    let(:student1) { create :student, course: course }
+    let(:lti_deployment) { create :lti_deployment, course: course }
+    # let(:lti_user) { create :lti_user, user: student1, lti_deployment: lti_deployment}
+    let!(:assessment) { create :assignment_with_criteria_and_results, course: course }
+    let!(:assessment2) { create :assignment_with_criteria_and_results, course: course }
+    before :each do
+      User.all.each { |usr| create :lti_user, user: usr, lti_client: lti_deployment.lti_client }
+      Result.joins(:grouping)
+            .joins(grouping: :assignment)
+            .where('assignment.id': assessment.id).update!(released_to_students: true)
+    end
+    it 'successfully gets grades for an assignment with released grades' do
+      expect(lti_deployment.get_assignment_marks(assessment)).not_to be_empty
+    end
+    it 'does not get unreleased grades' do
+      expect(lti_deployment.get_assignment_marks(assessment2)).to be_empty
+    end
+    it 'does not get unreleased grades for an assignment with released grades' do
+      Result.joins(:grouping)
+            .joins(grouping: :assignment)
+            .where('assignment.id': assessment2.id).first.update!(released_to_students: true)
+      expect(lti_deployment.get_assignment_marks(assessment2)).not_to be_empty
+    end
+  end
+  describe '#get_grade_entry_form_marks' do
+    let(:course) { create :course }
+    let(:student1) { create :student, course: course }
+    let(:lti_deployment) { create :lti_deployment, course: course }
+    # let(:lti_user) { create :lti_user, user: student1, lti_deployment: lti_deployment}
+    let(:assessment) { create :grade_entry_form_with_data, course: course }
+    before :each do
+      student1
```

Comment:
Use `let!` instead of `let` to force the creation of this object (and `assessment` on the next line)